src/bulk/common.ts

```diff
@@ -660,16 +665,39 @@ function handleMongoWriteConcernError(
  * @public
  * @category Error
  */
-export class BulkWriteError extends MongoError {
+export class MongoBulkWriteError extends MongoError {
   result?: BulkWriteResult;
 
-  /** Creates a new BulkWriteError */
+  /** Number of documents inserted. */
+  insertedCount?: number;
+  /** Number of documents matched for update. */
+  matchedCount?: number;
+  /** Number of documents modified. */
+  modifiedCount?: number;
+  /** Number of documents deleted. */
+  deletedCount?: number;
+  /** Number of documents upserted. */
+  upsertedCount?: number;
+  /** Inserted document generated Id's, hash key is the index of the originating operation */
+  insertedIds?: { [key: number]: ObjectId };
+  /** Upserted document generated Id's, hash key is the index of the originating operation */
+  upsertedIds?: { [key: number]: ObjectId };
+
+  /** Creates a new MongoBulkWriteError */
   constructor(error?: AnyError, result?: BulkWriteResult) {
     super(error as Error);
     Object.assign(this, error);
 
-    this.name = 'BulkWriteError';
+    this.name = 'MongoBulkWriteError';
     this.result = result;
+
+    this.insertedCount = result?.insertedCount;
+    this.matchedCount = result?.matchedCount;
+    this.modifiedCount = result?.modifiedCount || 0;
```

Comment:
Don't need to provide a default here, `modifiedCount` is already guaranteed to be a number in `BulkWriteResult`.